provider/gateway/rest/router_test.go

```diff
@@ -85,6 +102,187 @@ func runRouterTest(t *testing.T, authClient bool, fn func(*routerTest)) {
 	})
 }
 
+func testCertHelper(t *testing.T, test *routerTest) {
+	test.pmclient.On(
+		"Submit",
+		mock.Anything,
+		mock.AnythingOfType("types.DeploymentID"),
+		mock.AnythingOfType("manifest.Manifest"),
+	).Return(nil)
+
+	dseq := uint64(testutil.RandRangeInt(1, 1000))
+
+	uri, err := makeURI(test.host, submitManifestPath(dseq))
+	require.NoError(t, err)
+
+	sdl, err := sdl.ReadFile(testSDL)
+	require.NoError(t, err)
+
+	mani, err := sdl.Manifest()
+	require.NoError(t, err)
+
+	buf, err := json.Marshal(mani)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("PUT", uri, bytes.NewBuffer(buf))
+	require.NoError(t, err)
+
+	req.Header.Set("Content-Type", contentTypeJSON)
+
+	_, err = test.gclient.hclient.Do(req)
+	require.Error(t, err)
```

Comment:
It seems like we should assert some specific error here, or something in the text string at least.

Reply:
https://github.com/ovrclk/akash/pull/1036